src/frontends/common/include/openvino/frontend/place.hpp

```diff
@@ -57,11 +58,21 @@ namespace frontend {
 ///                    V
 ///                [Tensor C]
 ///
-class FRONTEND_API Place {
+class FRONTEND_API Place : protected SharedObjectExtension<Place> {
 public:
     typedef std::shared_ptr<Place> Ptr;
 
-    virtual ~Place() = 0;
+    /// \brief Place default ctor.
+    Place() = default;
+
+    /// * \brief Place ctor used when other Place is set as pimpl and shared object pointer to library required by
+    /// pimpl.
+    /// *
+    /// * \param pimpl          Input place as pimpl.
+    /// * \param shared_object  Pointer to shared object (library).
+    Place(Ptr pimpl, std::shared_ptr<void> shared_object);
```

Comment:
No, this ctor should be used instead Place(). But this class is as base for pimpl and class which wrap pimpl it the Place() is still required. 
Probably if there would be seprate Places bases form impl and wrapper it would be nicer but will require some bigger code refactoring.
Currently Place has no friends and this ctor is not only used by derived classes.

Reply:
If I remember right we are providing Place API for our end users, I don't think that we need to provide such API, because it is needed only for internal usage.